rpc/server.go

```diff
@@ -141,20 +139,22 @@ func (s *Server) ServeCodec(codec ServerCodec) {
 	s.codecsMu.Lock()
 	if atomic.LoadInt32(&s.run) != 1 { // server stopped
 		s.codecsMu.Unlock()
-		return
+		return &shutdownError{}
 	}
 	s.codecs.Add(codec)
```

Comment:
Isn't the codec removed on line 135? Or is this code added *after* this comment?